storage/interceptors/smaap_service_instance_interceptor.go

```diff
@@ -421,13 +442,28 @@ func (i *ServiceInstanceInterceptor) deleteSingleInstance(ctx context.Context, i
 	if bindingsCount, err = i.repository.Count(ctx, types.ServiceBindingType, byServiceInstanceID); err != nil {
 		return fmt.Errorf("could not fetch bindings for instance with id %s", instance.ID)
 	}
+
+	if instance.IsShared() {
+		referencesList, err := storage.GetInstanceReferencesByID(ctx, i.repository, instance.ID)
+		if err != nil {
```

Comment:
we use `referencesList` instead of `count` because we present this list to the client in case of an error.

```
if referencesList != nil && referencesList.Len() > 0 {
			return util.HandleReferencesError(util.ErrSharedInstanceHasReferences, types.ObjectListIDsToStringArray(referencesCount))
}
```

I think we should consider keeping it that way.
What do you think?